pkg/share/manager/sql/model/model.go

```diff
@@ -99,6 +99,95 @@ type ShareState struct {
 	Alias  string `gorm:"size:64"`
 }
 
+// OCM Shares - uses BaseModel to share common ShareID with other share types
+type OcmShares struct {
+	BaseModel
+	Token        string             `gorm:"size:255;not null;uniqueIndex"`
+	FileidPrefix string             `gorm:"size:64;not null;uniqueIndex:idx_fileid_source_share_with"`
+	ItemSource   string             `gorm:"size:64;not null;uniqueIndex:idx_fileid_source_share_with"`
+	Name         string             `gorm:"type:text;not null"`
+	ShareWith    string             `gorm:"size:255;not null;uniqueIndex:idx_fileid_source_share_with"`
+	Owner        string             `gorm:"size:255;not null;uniqueIndex:idx_fileid_source_share_with"`
+	Initiator    string             `gorm:"type:text;not null"`
+	Ctime        time.Time          `gorm:"not null"`
+	Mtime        time.Time          `gorm:"not null"`
+	Expiration   datatypes.NullTime `gorm:"default:null"`
+	Type         uint8              `gorm:"not null"`
+}
+
+// OCM Shares Access Methods
+type OcmSharesAccessMethods struct {
+	gorm.Model
+	OcmShareID uint      `gorm:"not null;uniqueIndex:idx_ocm_share_method"`
+	OcmShare   OcmShares `gorm:"constraint:OnDelete:CASCADE;foreignKey:OcmShareID;references:Id"`
+	Type       uint8     `gorm:"not null;uniqueIndex:idx_ocm_share_method"`
+}
+
+// WebDAV Access Method
+type OcmAccessMethodWebdav struct {
```

Comment:
Since we are redesigning the tables anyway, can we not just put this info in `OCMShareAccessMethods` anyway? Since `OcmAccessMethodWebapp` and `OcmAccessMethodWebdav` basically have the same columns?